### PR TITLE
Docs: Add event tracking for buttons

### PR DESCRIPTION
**Pull request by dangerismycat**
This PR adds Google Analytics event tracking to all our buttons across the docs. If there's a button I missed, please let me know!

We're logging the button clicked, and sometimes additional metadata where useful. You can check this by viewing the [real-time events](https://analytics.google.com/analytics/web/#/realtime/rt-event/a12967896w242172598p225758288/metric.type=5&filter.list=40==button;) and clicking around on the preview site.

image

### CODE REVIEW

File: docs/src/components/buttons/SidebarCategorizationButton.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  onClick: () => void,
+  sidebarOrganisedBy: SidebarOrganisedBy,
+|};
+
+export default function SidebarCategorizationButton({ onClick, sidebarOrganisedBy }: Props): Node {
+  const sidebarOrganisedByCopy =
+    sidebarOrganisedBy === 'categorized' ? 'Alphabetical' : 'Categorize';
+
+  return (
+    <Tooltip inline text={`Sidebar: ${sidebarOrganisedByCopy}`}>
```
Comment (AlbertCarreras) [inline docs/src/components/buttons/SidebarCategorizationButton.js:18]: I feel these Tooltips are not following best practices.

```
    <Tooltip inline text={`Toggle sidebar categorization from ${sidebarOrganisedByCopy} to ${sidebarOrganisedBy === 'categorized' ? 'Categorize' : 'Alphabetical'}`}>
        <IconButton
          accessibilityLabel=""
```

but I'd love to hear @ayeshakmaz thoughts,

File: docs/src/components/buttons/trackButtonClick.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// @flow strict
+
+// Since some buttons have text and some only have icons, we'll use the a11yLabel for consistency
```
Comment (AlbertCarreras) [inline docs/src/components/buttons/trackButtonClick.js:3]: We are using 'names', it's not always the exact same `a11yLabel`, right?

File: docs/src/components/buttons/OpenSandboxButton.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+type Props = {|
+  onClick: () => void,
+|};
+
+export default function OpenSandboxButton({ onClick }: Props): Node {
+  const label = 'Open in CodeSandbox';
+
+  return (
+    <Tooltip inline text={label} idealDirection="up">
```
Comment (AlbertCarreras) [inline docs/src/components/buttons/OpenSandboxButton.js:14]: This should be fixed to follow Tooltip best practices. They should not be repeated. We can omit `accessibilityLabel=""`

File: docs/src/components/buttons/LTRButton.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    __path:
+      textDirection === 'rtl'
+        ? 'M9 10v5h2V4h2v11h2V4h2V2H9C6.79 2 5 3.79 5 6s1.79 4 4 4zm12 8l-4-4v3H5v2h12v3l4-4z'
+        : 'M10 10v5h2V4h2v11h2V4h2V2h-8C7.79 2 6 3.79 6 6s1.79 4 4 4zm-2 7v-3l-4 4 4 4v-3h12v-2H8z',
+  };
+
+  const directionCopy = textDirection === 'rtl' ? 'Left-To-Right View' : 'Right-To-Left View';
+
+  return (
+    <Tooltip inline text={directionCopy}>
```
Comment (AlbertCarreras) [inline docs/src/components/buttons/LTRButton.js:22]: Same here as comments below, Tooltip + a11yLabel should work together.

File: docs/src/components/buttons/DarkModeButton.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  onClick: () => void,
+|};
+
+export default function DarkModeButton({ colorScheme, onClick }: Props): Node {
+  const colorSchemeCopy = colorScheme === 'light' ? 'Dark-Mode View' : 'Light-Mode View';
+
+  return (
+    <Tooltip inline text={colorSchemeCopy}>
+      <IconButton
+        accessibilityLabel="Toggle color scheme: light / dark mode views"
```
Comment (AlbertCarreras) [inline docs/src/components/buttons/DarkModeButton.js:19]: Same as comments below

File: docs/src/components/buttons/CopyCodeButton.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  onClick: () => void,
+|};
+
+export default function CopyCodeButton({ onClick }: Props): Node {
+  const label = 'Copy code';
+
+  return (
+    <Tooltip inline text={label} idealDirection="up">
+      <IconButton
+        accessibilityLabel={label}
```
Comment (AlbertCarreras) [inline docs/src/components/buttons/CopyCodeButton.js:16]: `a11yLabel=""` shouldn't be repeated

File: docs/src/components/buttons/CollapseExpandCodeButton.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import trackButtonClick from './trackButtonClick.js';
+
+type Props = {|
+  expanded: boolean,
+  name: string,
+  onClick: () => void,
+|};
+
+export default function CollapseExpandCodeButton({ expanded, name, onClick }: Props): Node {
+  const label = `${expanded ? 'Collapse' : 'Expand'} code example`;
```
Comment (AlbertCarreras) [inline docs/src/components/buttons/CollapseExpandCodeButton.js:13]: Same, repeated text.

File: docs/src/components/PageHeader.js [hunk 8]
```diff
@@ -52,7 +53,11 @@ export default function ComponentHeader({
             ) : null}
           </Heading>
           {showSourceLink && (
-            <a href={githubUrl(fileName ?? name)} target="blank">
+            <a
```
Comment (AlbertCarreras) [inline docs/src/components/PageHeader.js:56]: Could we use Link here?

File: docs/src/components/HeaderMenu.js [hunk 9]
```diff
@@ -1,76 +1,58 @@
...
+      <SidebarCategorizationButton
+        onClick={() =>
+          setSidebarOrganizedBy(
+            sidebarOrganisedBy === 'categorized' ? 'alphabetical' : 'categorized',
+          )
+        }
+        sidebarOrganisedBy={sidebarOrganisedBy}
+      />
+
+      <Tooltip inline text="Opens CodeSandbox ready to start coding with Gestalt">
```
Comment (AlbertCarreras) [inline docs/src/components/HeaderMenu.js:38]: Could this one also be a separate button?

File: docs/src/components/HeaderMenu.js [hunk 10]
```diff
@@ -1,76 +1,58 @@
...
+            href="https://codesandbox.io/s/k5plvp9v8v"
+            onClick={() => trackButtonClick('Playground')}
+            target="blank"
+          >
             <Box padding={2}>Playground</Box>
           </GestaltLink>
         </Text>
       </Tooltip>
+
       <Text color="white">
```
Comment (AlbertCarreras) [inline docs/src/components/HeaderMenu.js:50]: Can we change order here?

         <GestaltLink
          href="https://github.com/pinterest/gestalt"
          onClick={() => trackButtonClick('GitHub')}
          target="blank"
        > 
          <Box padding={2}>\
               <Text color="white">GitHub</Text>
         </Box>
        </GestaltLink>

Comment (AlbertCarreras) [reply docs/src/components/HeaderMenu.js:50]: One above too.

**Comment by AlbertCarreras** [comment]
I think Copy Link is missing. http://localhost:8888/Tooltip#Labels

File: docs/src/components/buttons/SidebarCategorizationButton.js (hunk 1 above)
Comment (dangerismycat) [reply docs/src/components/buttons/SidebarCategorizationButton.js:18]: That's possible, but out of scope for this PR. This is just adding tracking

File: docs/src/components/buttons/OpenSandboxButton.js (hunk 3 above)
Comment (dangerismycat) [reply docs/src/components/buttons/OpenSandboxButton.js:14]: out of scope

File: docs/src/components/buttons/LTRButton.js (hunk 4 above)
Comment (dangerismycat) [reply docs/src/components/buttons/LTRButton.js:22]: out of scope

File: docs/src/components/buttons/DarkModeButton.js (hunk 5 above)
Comment (dangerismycat) [reply docs/src/components/buttons/DarkModeButton.js:19]: out of scope

File: docs/src/components/buttons/CopyCodeButton.js (hunk 6 above)
Comment (dangerismycat) [reply docs/src/components/buttons/CopyCodeButton.js:16]: out of scope

File: docs/src/components/buttons/CollapseExpandCodeButton.js (hunk 7 above)
Comment (dangerismycat) [reply docs/src/components/buttons/CollapseExpandCodeButton.js:13]: out of scope

File: docs/src/components/PageHeader.js (hunk 8 above)
Comment (dangerismycat) [reply docs/src/components/PageHeader.js:56]: out of scope

File: docs/src/components/HeaderMenu.js (hunk 10 above)
Comment (dangerismycat) [reply docs/src/components/HeaderMenu.js:50]: out of scope

File: docs/src/components/buttons/SidebarCategorizationButton.js (hunk 1 above)
Comment (ayeshakmaz) [reply docs/src/components/buttons/SidebarCategorizationButton.js:18]: There's a bug filed for Tooltips and A11Y that will affect this anyway, so worth waiting until that is fixed

Comment (AlbertCarreras) [reply docs/src/components/buttons/SidebarCategorizationButton.js:18]: @ayeshakmaz what's the bug number?

Comment (AlbertCarreras) [reply docs/src/components/buttons/SidebarCategorizationButton.js:18]: Also, we recommend that in best practices... so not quite sure how bug + best practices intersect.

**Comment by AlbertCarreras** [comment]
LGTM besides my comments. I'll R+ cos it tracks and that's the PR scope.

**Review by AlbertCarreras — APPROVED** [review]

File: docs/src/components/buttons/SidebarCategorizationButton.js (hunk 1 above)
Comment (ayeshakmaz) [reply docs/src/components/buttons/SidebarCategorizationButton.js:18]: [PDS-2543 ](https://jira.pinadmin.com/browse/PDS-2543)

**Pull request merged**